### PR TITLE
remove some redundant fields from MempoolItem

### PR DESCRIPTION
**Pull request by arvidn**
Apart from lowering memory footprint of the mempool, the main objective of this change is to (slightly) improve the performance of serializing and deserialializing `MempoolItem`s. A forthcoming PR storing these in a sqlite database rely in serializing and deserializing these.

This change shaves off a small amount of time deserializing:

| before | after | after / before |
| --- | --- | --- |
| 131.7 | 120.68 | 91.6 % |

before:
```
Profiling create_bundle_from_mempool()
  output written to: mempool-create-mt.png
  time: 65.8534s
  per call: 131.71ms
```

after:
```
Profiling create_bundle_from_mempool()
  output written to: mempool-create-mt.png
  time: 60.3389s
  per call: 120.68ms
```

### CODE REVIEW

File: chia/types/mempool_item.py [hunk 1]
```diff
@@ -36,6 +35,15 @@ def fee_per_cost(self) -> float:
     def name(self) -> bytes32:
         return self.spend_bundle_name
 
+    @property
+    def cost(self) -> uint64:
+        assert self.npc_result.conds is not None
+        return uint64(self.npc_result.conds.cost)
+
+    @property
+    def additions(self) -> List[Coin]:
```
Comment (AmineKhaldi) [inline chia/types/mempool_item.py:44]: This alters some RPCs as they won't include these in their responses anymore.

File: chia/types/mempool_item.py [hunk 2]
```diff
@@ -36,6 +35,15 @@ def fee_per_cost(self) -> float:
...
+        return additions_for_npc(self.npc_result)
```
Comment (AmineKhaldi) [inline chia/types/mempool_item.py:45]: I wonder if we can include additions (and removals) in `NPCResult` instead of computing on the python side.

Comment (arvidn) [reply chia/types/mempool_item.py:45]: removals are already included in both `SpendBundle` and `SpendBundleConditions`.
additions are already included in `SpendBundleConditions`, available by iterating each `Spend`

File: chia/types/mempool_item.py (hunk 1 above)
Comment (arvidn) [reply chia/types/mempool_item.py:44]: you have a PR to address that, do you think we could get that landed?

Comment (AmineKhaldi) [reply chia/types/mempool_item.py:44]: I think so: https://github.com/Chia-Network/chia-blockchain/pull/14476

File: chia/types/mempool_item.py (hunk 2 above)
Comment (AmineKhaldi) [reply chia/types/mempool_item.py:45]: Right, but I meant the need for `additions_for_npc` and any similar computation functions on the python side, on `NPCResult`.

**Review by aqk — APPROVED** [review]
This is good - I've certainly wanted the conditions for a MempoolItem before.

File: chia/types/mempool_item.py (hunk 2 above)
Comment (arvidn) [reply chia/types/mempool_item.py:45]: `additions_for_npc` is cheap, it just loops over the `spends` and appends the `create_coin` lists together.

**Review by AmineKhaldi — APPROVED** [review]

**Review by trepca — APPROVED** [review]

**Pull request merged**